server.py

```diff
@@ -32,7 +33,30 @@ def get_shopcarts(id):
 
     return jsonify(message), rc
 
-
+######################################################################
+# Create a Shopcart
+######################################################################
+@app.route('/shopcarts/<int:id>', methods=['PUT'])
```

Comment:
You've made this a `PUT` request, but I think it should be `POST`, since we're creating an entirely new shopcart, as discussed in #39 . Same goes for the unit tests.